lib/rubygems/commands/pristine_command.rb

```diff
@@ -104,6 +104,8 @@ def execute
               end.flatten
             end
 
+    specs = specs.select{|spec| spec.platform == RUBY_ENGINE }
```

Comment:
I see that normally the check for the current platform is done using `Platform.local`. For example, 

https://github.com/rubygems/rubygems/blob/8d5490364ac3a50eecefe880cbf1201ba355dfb0/lib/rubygems/resolver/installer_set.rb#L65

or

https://github.com/rubygems/rubygems/blob/8d5490364ac3a50eecefe880cbf1201ba355dfb0/lib/rubygems/resolver/installer_set.rb#L65

Should the same check be applied here or it's better to use `RUBY_ENGINE`?

Reply:
Comparisons of `s.platform` also involves the issue of how the `--platform` option affects gem selection.  There seems to be an assumption that `--platform=ruby` is the only valid use, probably based on the fact that platform specific gems are chosen before 'ruby' gems.